service/src/test/java/com/epam/rft/atsy/service/impl/StatesServiceImplTest.java

```diff
@@ -337,22 +336,41 @@ private void assertCandidateApplicationDTOCollection(Collection<CandidateApplica
         assertThat(candidateApplicationDTOCollection, equalTo(expectedCollection));
     }
 
-    private List<CandidateApplicationDTO> getExpectedCandidateApplicationDTOListFromApplicationEntityList(List<ApplicationEntity> applicationEntityList) {
-        List<CandidateApplicationDTO> candidateApplicationDTOList = new LinkedList<>();
-
-        for (ApplicationEntity applicationEntity : applicationEntityList) {
-            StateEntity stateEntity = statesRepository.findTopByApplicationEntityOrderByStateIndexDesc(applicationEntity);
-
-            CandidateApplicationDTO candidateApplicationDTO = CandidateApplicationDTO.builder()
-                    .applicationId(applicationEntity.getId())
-                    .stateType(stateEntity.getStateType())
-                    .positionName(applicationEntity.getPositionEntity().getName())
-                    .lastStateId(stateEntity.getId())
-                    .creationDate(SIMPLE_DATE_FORMAT.format(applicationEntity.getCreationDate()))
-                    .modificationDate(SIMPLE_DATE_FORMAT.format(stateEntity.getCreationDate()))
-                    .build();
-            candidateApplicationDTOList.add(candidateApplicationDTO);
-        }
-        return candidateApplicationDTOList;
+    private void assertStateEntityWhenSavingStateEntity(StateEntity stateEntity, StateEntity expectedSavedStateEntity) {
+        assertThat(stateEntity, notNullValue());
+        assertThat(stateEntity.getApplicationEntity(), equalTo(expectedSavedStateEntity.getApplicationEntity()));
+        assertThat(stateEntity.getCreationDate(), greaterThanOrEqualTo(expectedSavedStateEntity.getCreationDate()));
+    }
+
+    private StateEntity getExpectedSavedStateEntity() {
```

Comment:
I don't see your logic when to place a test data into a field initialized by the setup method or into a separate method, which is called from the test cases.

This comment is also apply for the next two methods.